pages/management/PKPToolsHandler.inc.php

```diff
@@ -160,4 +166,101 @@ public function resetPermissions($args, $request)
         echo json_encode(true);
         die;
     }
+
+    public function jobs($args, $request)
+    {
+        $this->setupTemplate($request, true);
+
+        $templateMgr = TemplateManager::getManager($request);
+
+        $queuedJobs = $this->buildQueuedJobsTable();
+
+        $totalQueuedJobs = $queuedJobs['total'];
+        $queuedData = $queuedJobs['data'];
+
+        $state = [
+            'components' => [
+                $queuedData->id => $queuedData->getConfig(),
+            ]
+        ];
+
+        $templateMgr->setState($state);
+
+        $templateMgr->assign([
+            'pageComponent' => 'JobsPage',
+            'pageTitle' => __('navigation.tools.jobs'),
+            'totalQueuedJobs' => $totalQueuedJobs
+        ]);
+
+        $templateMgr->display('management/tools/jobs.tpl');
+    }
+
+    /**
+     * Build the Queued Jobs Table
+     */
+    protected function buildQueuedJobsTable(): array
+    {
+        $dateFormatShort = Application::get()
+            ->getRequest()
+            ->getContext()
+            ->getLocalizedDateTimeFormatShort();
+
+        $jobsQuery = DB::table('jobs')
```

Comment:
The handler class shouldn't be building SQL queries; this should be in a DAO or Collector.

Reply:
(Likewise, the handler class shouldn't be iterating through raw database records.)

Reply:
Unfortunately, at this point of code history, we doesn't have a Domain/DAO/Repository/Collector for the `jobs` table.

I've created a Eloquent Model for `jobs` at #4622. It was two months before the starting of Repository refactoring.

Right now I don't know if I should wait for the #4622 being merged, with his Eloquent Model stuff, or create a Repository here for the `jobs`. Both patterns are conflicting, in my view.